cram/cram_encode.c

```diff
@@ -1825,8 +1843,11 @@ int cram_encode_container(cram_fd *fd, cram_container *c) {
             // This starts as 'N' and is amended on-the-fly as we go
             // based on MD:Z tags.
             if ((c->ref_id = bam_ref(b)) >= 0) {
-                c->ref_free = 1;
                 c->ref = NULL;
+                // c->ref_ref is boolean; whether to free c->ref.  In this
+                // case c->ref will be our auto-embedded sequence.
+                // Do not confuse with fd->ref_free which is a ptr to free
```

Comment:
Slightly ambiguous — is this “a ptr to be freed” or ”`fd->ref_free` points to [something named] `free`”?

Reply:
I'll edit.  I was I admit being somewhat deliberately terse to avoid it going over 80 columns :)

Reply:
A very valid viewpoint 😄 though I have to admit sometimes I've restandardised on 96.

Reply:
Indeed sometimes 80 is a bit restrictive.  It does need a limit somewhere though.  Sometimes it gets rediculous.  Htslib's longest line is 245 (an error string which could be broken up with `"blah""blah"` syntax),   Similarly all of samtools longest lines are printfs in stats.c, and in bcftools it's VCF printing in csq.c.  Htslib's average line length is < samtools < bcftools too, as I expected.

The longest lines without quotes though are
https://github.com/samtools/samtools/blob/develop/reset.c#L364
https://github.com/samtools/htslib/blob/develop/samples/mpileup.c#L152
https://github.com/samtools/samtools/blob/develop/stats.c#L1518
https://github.com/samtools/htslib/blob/develop/hts.c#L4035

I find such source just hard to read as they break the indentation once things get line-wrapped in the terminal, or without line-wrapping (eg github) they're equally a pain as you have to keep scrolling back and forth.  Doubly bad in github due to having the horizontal scrollbar needing vertical scrolling to get to!  (Can use arrow keys instead)